### ISSUE TITLE
[fpga, pinmux] One issue to fpga debug interface not working

### ISSUE THREAD

**Issue by tjaychen** [lowRISC/opentitan#7783]
There have been reports of fpga debug not working popping up.

This is just one potential issue, possibly more. 

The jtag function now needs to be selected based on [tap selection pins](https://github.com/lowRISC/opentitan/blob/master/hw/ip/pinmux/rtl/pinmux_strap_sampling.sv#L180).

However, as part of our fpga debug setup, we are not currently selecting the tap select.
However, even if we did, one of the [tap select pins](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/data/top_earlgrey.hjson#L1337) is removed from [nexysvideo](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/data/top_earlgrey.hjson#L1311).

So we probably need to select another pin, and setup the configuration script to set these GPIOs.

**Comment by tjaychen** [comment]
@imphil 
@tomroberts-lowrisc 
@msfschaffner 

FYI. I took a quick look. I'm not saying this will fix the issue, but just a first pass looking over the code.

**Comment by msfschaffner** [comment]
Technically, the current pinmux configuration should work even with one stubbed-out TAP strap (at least when I added that, JTAG access still worked).

I.e., in order to select the RISC-V TAP, only strap 1 needs to be accessible due to the [strap encoding](https://docs.opentitan.org/hw/ip/pinmux/doc/#strap-sampling-and-tap-isolation).
Also, we currently run the FPGA image in the RMA life cycle state, such that the TAP straps are sampled continuously.

So I am not entirely sure this is because of an I/O issue or an issue with the debug module / debug software itself.

**Comment by msfschaffner** [comment]
@imphil I thought we had a simple JTAG test in our FPGA CI that checks whether the debugger can connect to the RISC-V TAP? Has that been removed?

**Comment by tjaychen** [comment]
i think we only have it for verilator (not for fpga).

**Comment by msfschaffner** [comment]
ah yeah you are right. if that test passes in Verilator, this should indicate that the pinmux strap sampling logic itself is working correctly (and that the device is in the correct life cycle state). 

let's wait for more info on the other bug.
it should be possible to at least connect to the RV DM via JTAG. 
so if that is not possible, this might indicate that there is another issue with the setup (e.g. wrong USB port used for the JTAG connection).

**Comment by tjaychen** [comment]
ah i forgot to point out [this](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/dv/verilator/chip_sim_tb.sv#L92).
I think all this stuff just gets bypassed for verilator.  It's part of the stuff we need to fix.

**Comment by tjaychen** [comment]
and like i mentioned, i think our scripts right now don't properly setup the tap selects anyways.

**Comment by msfschaffner** [comment]
ah yes fair enough.
yes the Verilator target needs some alignment...

**Comment by msfschaffner** [comment]
Carrying over two comments from #7759 for tracking purposes.

------------------------------

if you look at the linked issue, you'll see the issue is two fold

- the current configuration doesn't actually allocate a pin necessary to select the jtag interface
- our scripts right now i do not believe drive the appropriate signals needed to select the jtag interface.

So there is both a design change and a scripting change needed.

_Originally posted by @tjaychen in https://github.com/lowRISC/opentitan/issues/7759#issuecomment-901313880_

------------------------------

It could be one of the points that Tim has mentioned - but technically the pin mapping should be fine if only the RISC-V TAP needs to be selected (we would need another input pin to select other TAPs in the system, though). The scripts should be able to handle this as well, since even before the new TAP selection mechanism it was necessary to drive a "JTAG select signal" to the same pin (IOB7).

So it looks like some more debugging is needed to figure out the root cause of this issue...

_Originally posted by @msfschaffner in https://github.com/lowRISC/opentitan/issues/7759#issuecomment-901395960_

**Comment by tjaychen** [comment]
yeah after having a closer look, and i don't actually see anything wrong in this area. I retract my earlier statement. Still looking around..
